terraform-modules/aws/msk/main.tf

```diff
@@ -0,0 +1,157 @@
+  tags   = var.tags 
+}
+
+data "aws_iam_policy_document" "acmpca_bucket_access" {
+  count = var.s3_bucket_create ? 1 : 0
+  statement {
+    actions = [
+      "s3:GetBucketAcl",
+      "s3:GetBucketLocation",
+      "s3:PutObject",
+      "s3:PutObjectAcl",
+    ]
+
+    resources = [
+      aws_s3_bucket.this[0].arn,
+      "${aws_s3_bucket.this[0].arn}/*",
+    ]
+
+    principals {
+      identifiers = ["acm-pca.amazonaws.com"]
+      type        = "Service"
+    }
+  }
+}
+
+resource "aws_s3_bucket_policy" "this" {
+  count = var.s3_bucket_create ? 1 : 0
+  bucket = aws_s3_bucket.this[0].id
+  policy = data.aws_iam_policy_document.acmpca_bucket_access[0].json
+}
+
+#######################################
+# Private CA
+#######################################
+data "aws_partition" "current" {
+}
+
+resource "aws_acmpca_certificate_authority_certificate" "cacert" {
+  count = var.create_private_ca ? 1 : 0
+  certificate_authority_arn = aws_acmpca_certificate_authority.this[0].arn
+
+  certificate       = aws_acmpca_certificate.cert[0].certificate
+  certificate_chain = aws_acmpca_certificate.cert[0].certificate_chain
+}
+
+resource "aws_acmpca_certificate" "cert" {
+  count = var.create_private_ca ? 1 : 0
+  certificate_authority_arn   = aws_acmpca_certificate_authority.this[0].arn
+  certificate_signing_request = aws_acmpca_certificate_authority.this[0].certificate_signing_request
+  signing_algorithm           = "SHA512WITHRSA"
+
+  template_arn = "arn:${data.aws_partition.current.partition}:acm-pca:::template/RootCACertificate/V1"
+
+  validity {
+    type  = "YEARS"
+    value = local.years_valid
+  }
+}
+
+resource "aws_acmpca_certificate_authority" "this" {
+  count = var.create_private_ca ? 1 : 0
+  certificate_authority_configuration {
+    key_algorithm     = var.key_algorithm
+    signing_algorithm = var.signing_algorithm
+
+    subject {
+      common_name = var.common_name
+    }
+  }
+  
+  type = "ROOT"
+
+  revocation_configuration {
+    crl_configuration {
+      custom_cname       = "crl.${var.common_name}"
+      enabled            = true
+      expiration_in_days = var.expiration_in_days
+      s3_bucket_name     = var.s3_logs_bucket
+    }
+  }
+  
+  tags   = var.tags
+
+  depends_on = [aws_s3_bucket_policy.this]
+}
+
+#######################################
+# MSK Cluster
+#######################################
+module "msk" {
+  #Fixing Deprecated Field in Terraform for AWS MSK Apache Kafka Deployment
+  #----------------------------------------
+  # Why we are poiting to a fixed branch here?
+  # We are currently targeting a fixed branch in our code because we are waiting for authorization of a pull request (PR) in the provider 
+  # https://github.com/cloudposse/terraform-aws-msk-apache-kafka-cluster/pull/76 This branch is being used by client EP.
+  # In addition, this pull request repairs a warning regarding the deprecated field 'ebs_volume_size' in Terraform, as it was deprecated by AWS.
```

Comment:
This comment is outdated and should be removed?

Reply:
comments removed